nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,11 +423,215 @@ async fn run() -> Result<()> {
+        }
+    }
+    // This is when we are going to retry because there is not a single
+    // answer occurs more than any others
+    let mut accepted_result: Option<String> = None;
+    if max_count == 1 {
+        accepted_result = Some(answer);
+    }
+    let (result, winners) = get_winners(tasks, accepted_result);
+    // TODO(wangjun.hong): Replace below with a single update
+
+    if !winners.is_empty() && !result.is_none() {
+        reward(assign_event_id.clone(), &winners, result.unwrap(), classifier_result_collection,
+        collection, 1).await;
+    }
+
+    Ok(())
+}
+
+/**
+Go through all tasks and return all winners and the result payload
+When result is None, no winners are returned
+**/
+fn get_winners(
+    tasks: &Vec<AssignerTask>,
+    result: Option<String>
+) -> (Option<ResultPayload>, Vec<Sender>) {
+    tracing::debug!("Result is: {:?}", result);
+    let mut winners: Vec<Sender> = vec![];
+    let mut result_payload: Option<ResultPayload> = None;
+    tasks.iter().for_each(|task| {
+        let task_result = get_task_identifier(task);
+        if task_result == result && !task_result.is_none() {
+            winners.push(task.worker.clone());
+            result_payload = Some(task.result.clone());
+        }
+    });
+
+    (result_payload, winners)
+}
+
+// Return values from a task identifier, in case of
+// failure happened, return None
+fn get_task_identifier(
+    task: &AssignerTask
+) -> Option<String> {
+    match get_result_identifier(task.result.output.clone()) {
+        Ok(Some(val)) => {
+            Some(val)
+        }
+        Ok(None) => {
+            None
+        }
+        Err(err) => {
+            tracing::debug!("Failed to parse identifier, error: {}", err);
+            None
+        }
+    }
+}
+
+async fn reward(
+    assign_event_id: String,
+    workers: &Vec<Sender>,
+    payload: ResultPayload,
+    classifier_result_collection: &Collection<ClassifierResult>,
+    collection: &Collection<FinishedJobs>,
+    job_type: u16,
+) -> bool {
+    tracing::debug!(
+        "All the results for {} are consistent, sending",
+        assign_event_id
+    );
+    let raw_data_id = payload.header.raw_data_id.clone();
+    let result_identifier = get_result_identifier(payload.output.clone());
+    match result_identifier   {
+        Err(_) =>{},
+        Ok(None)=>{},
+        Ok(Some(tag_id)) =>{
+            // Save classifier result to db
+            let classifier_result = ClassifierResult {
+                _id: raw_data_id,
+                kv_key: payload.kv_key.clone(),
+                tag_id
+            };
+            match classifier_result_collection.insert_one(classifier_result, None).await {
+                Ok(_) => {
+                    tracing::info!("Write classifier result to db: {}", payload.kv_key.clone());
+                }
+                Err(err) => {
+                    tracing::error!("Cannot write classifier result to db: {}", err);
```

Comment:
If we can write the classifier result and then, later, we cannot write it to the finished jobs collection, we increment FAILED_JOBS.
But if we fail to write the classifier result, we don't do the same..